tutorials/scripting/c_sharp/c_sharp_basics.rst

```diff
@@ -292,7 +292,42 @@ the performance of C# in Godot — while generally in the same order of magnitud
 — is roughly **~4×** that of GDScript in some naive cases. C++ is still
 a little faster; the specifics are going to vary according to your use case.
 GDScript is likely fast enough for most general scripting workloads.
-C# is faster, but requires some expensive marshalling when talking to Godot.
+
+Most properties of Godot C# objects that are based on ``Godot.Object``
+(e.g. any ``Node`` like ``Control`` or ``Node3D`` like ``Camera3D``) require some
+expensive marshalling as they are talking to Godot's C++ core.
+Any code that needs to access or modify such properties multiple times should assign values into a local
+variable and overwrite their values only once if possible:
```

Comment:
I'm not convinced about the "should" here. I think something like "consider doing this instead" would be better.